Development/nmos/control_protocol_state.cpp

```diff
@@ -0,0 +1,360 @@
+#include "nmos/control_protocol_state.h"
+
+#include "nmos/control_protocol_methods.h"
+#include "nmos/control_protocol_resource.h"
+
+namespace nmos
+{
+    namespace experimental
+    {
+        namespace details
+        {
+            // create control class
+            //   where
+            //     properties: vector of NcPropertyDescriptor can be constructed using make_control_class_property
```

Comment:
Comment feels ambiguous; is it an NcPropertyDescriptor or _vector_ of NcPropertyDescriptor that is created by make_control_class_property?
```suggestion
            //     properties: vector of NcPropertyDescriptor where NcPropertyDescriptor can be constructed using make_control_class_property
```